courses/models.py

```diff
@@ -603,7 +603,9 @@ class CourseRun(TimestampedModel):
     )
     product = GenericRelation(Product, related_query_name="course_run")
     title = models.CharField(max_length=255)
-    courseware_id = models.CharField(max_length=255, unique=True)
+    courseware_id = models.CharField(
+        max_length=255, unique=True, validators=[validate_url_path_field]
+    )
```

Comment:
We have added the Validator to avoid bad IDs in the Future, We haven't added a data migration to fix the existing bad Courseware IDs. Is that intentional?

Reply:
Yes this is intentional, If we update any pervious courseware IDs we might break previous orders or receipts.

Reply:
Thanks for the Answers. PR Looks Good. Approving!